hudi-utilities/src/test/java/org/apache/hudi/utilities/functional/TestDeltaStreamerWithOverwriteLatestAvroPayload.java

```diff
@@ -0,0 +1,92 @@
+
+  @BeforeAll
+  public static void initClass() throws Exception {
+    UtilitiesTestBase.initClass(true);
+    PARQUET_SOURCE_ROOT = dfsBasePath + "/parquetFiles";
+
+    // prepare the configs.
+    UtilitiesTestBase.Helpers.copyToDFS("delta-streamer-config/base.properties", dfs, dfsBasePath + "/base.properties");
+    UtilitiesTestBase.Helpers.copyToDFS("delta-streamer-config/sql-transformer.properties", dfs,
+        dfsBasePath + "/sql-transformer.properties");
+    UtilitiesTestBase.Helpers.copyToDFS("delta-streamer-config/source.avsc", dfs, dfsBasePath + "/source.avsc");
+    UtilitiesTestBase.Helpers.copyToDFS("delta-streamer-config/source-flattened.avsc", dfs, dfsBasePath + "/source-flattened.avsc");
+    UtilitiesTestBase.Helpers.copyToDFS("delta-streamer-config/target.avsc", dfs, dfsBasePath + "/target.avsc");
+  }
+
+  @Test
+  public void testOverwriteLatestAvroPayload() throws Exception {
+    // test defaultDeleteField
+    this.testOverwriteLatestAvroPayload(null);
+
+    // test userDefinedDeleteField
+    this.testOverwriteLatestAvroPayload("user_defined_delete_field");
+  }
+
+  public void testOverwriteLatestAvroPayload(String deleteMarkerField) throws Exception {
+    String path = PARQUET_SOURCE_ROOT + "/1.parquet";
+    List<GenericRecord> records = HoodieTestDataGenerator.genericRecords(5, false, 0);
+    Helpers.saveParquetToDFS(records, new Path(path));
+
+    TypedProperties parquetProps = new TypedProperties();
+    parquetProps.setProperty("include", "base.properties");
+    parquetProps.setProperty("hoodie.datasource.write.recordkey.field", "_row_key");
+    parquetProps.setProperty("hoodie.datasource.write.partitionpath.field", "not_there");
+    parquetProps.setProperty("hoodie.deltastreamer.source.dfs.root", PARQUET_SOURCE_ROOT);
+    if (deleteMarkerField != null) {
+      parquetProps.setProperty(HoodieWriteConfig.DELETE_FIELD_PROP, deleteMarkerField);
+    }
+    Helpers.savePropsToDFS(parquetProps, dfs, dfsBasePath + "/" + PROPS_FILENAME_TEST_PARQUET);
+
+    String tableBasePath = dfsBasePath + "/test_overwrite_lastest_avro_payload_table";
+
+    HoodieDeltaStreamer deltaStreamer = new HoodieDeltaStreamer(
+        TestHoodieDeltaStreamer.TestHelpers.makeConfig(tableBasePath, HoodieDeltaStreamer.Operation.INSERT, ParquetDFSSource.class.getName(),
+            null, PROPS_FILENAME_TEST_PARQUET, false,
+            false, 100000, false, null, null, "timestamp"), jsc);
+    deltaStreamer.sync();
+    TestHoodieDeltaStreamer.TestHelpers.assertRecordCount(5, tableBasePath + "/*/*.parquet", sqlContext);
+
+    String path2 = PARQUET_SOURCE_ROOT + "/2.parquet";
+    List<GenericRecord> records2 = HoodieTestDataGenerator.genericRecords(2, true, 1);
+    Helpers.saveParquetToDFS(records2, new Path(path2));
+    deltaStreamer.sync();
+
+    TestHoodieDeltaStreamer.TestHelpers.assertRecordCount(3, tableBasePath + "/*/*.parquet", sqlContext);
```

Comment:
is it possible to verify the records for equality ?

Reply:
Since the returned Spark Row is not easy to compare with GenericRecord, it is easier for us to compare only the key.